src/android/BluetoothLePlugin.java

```diff
@@ -1480,7 +1480,12 @@ private void connectAction(JSONArray args, CallbackContext callbackContext) {
 
     connections.put(device.getAddress(), connection);
 
-    BluetoothGatt bluetoothGatt = device.connectGatt(cordova.getActivity().getApplicationContext(), autoConnect, bluetoothGattCallback);
+    BluetoothGatt bluetoothGatt = null;
+    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
+      bluetoothGatt = device.connectGatt(cordova.getActivity().getApplicationContext(), autoConnect, bluetoothGattCallback, BluetoothDevice.TRANSPORT_LE);
```

Comment:
I'm slightly concerned with this breaking behavior for people who relied on TRANSPORT_AUTO. Could you make this a parameter instead? If version >= M and a transport has been specified, call #connectGatt() with a transport type.

Reply:
Please check out the updated version